### PR TITLE
P2893R3 Variadic friends

### LINKED ISSUE #6871: [2024-03 CWG Motion 10] Variadic friends
https://wiki.edg.com/pub/Wg21tokyo2024/StrawPolls/P2893R3.html

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #6871 
Fixes cplusplus/papers#1567

### CODE REVIEW

File: source/classes.tex [hunk 1]
```diff
@@ -4841,7 +4865,7 @@
 
 \pnum
 A friend declaration that does not declare a function
-shall have one of the following forms:
+shall be a \grammarterm{friend-type-declaration}.
 
 \begin{ncsimplebnf}
```
Comment (cpplearner) [inline source/classes.tex:4870]: This BNF should be deleted.

Comment (jensmaurer) [reply source/classes.tex]: Yes. Fixed.

File: source/preprocessor.tex [hunk 2]
```diff
@@ -1869,6 +1869,7 @@
 \defnxname{cpp_unicode_literals}                  & \tcode{200710L} \\ \rowsep
 \defnxname{cpp_user_defined_literals}             & \tcode{200809L} \\ \rowsep
 \defnxname{cpp_using_enum}                        & \tcode{201907L} \\ \rowsep
+\defnxname{cpp_variable_friend}                   & \tcode{202403L} \\ \rowsep
 \defnxname{cpp_variable_templates}                & \tcode{201304L} \\ \rowsep
```
Comment (frederick-vs-ja) [inline source/preprocessor.tex:1873]: This looks like a typo.
```suggestion
\defnxname{cpp_variable_templates}                & \tcode{201304L} \\ \rowsep
\defnxname{cpp_variadic_friend}                   & \tcode{202403L} \\ \rowsep
```

Comment (jensmaurer) [reply source/preprocessor.tex]: fixed

**Pull request merged**